src/boot/bootctl.c

```diff
@@ -73,49 +73,56 @@ static int verify_esp(
         struct statfs sfs;
         sd_id128_t uuid = SD_ID128_NULL;
         uint32_t part = 0;
+        bool quiet;
         int r;
 
         assert(p);
 
+        /* Non-root user can run only `bootctl status`, then if error occured in the following, it does not cause any issues.
+         * So, let's silence the error messages. */
+        quiet = (geteuid() != 0);
+
         if (statfs(p, &sfs) < 0) {
 
                 /* If we are searching for the mount point, don't generate a log message if we can't find the path */
                 if (errno == ENOENT && searching)
                         return -ENOENT;
 
-                return log_error_errno(errno, "Failed to check file system type of \"%s\": %m", p);
+                return quiet ? -errno : log_error_errno(errno, "Failed to check file system type of \"%s\": %m", p);
```

Comment:
It's usually better to log at debug level (it makes any debugging much easier). And the condition could be made much stricter. Why not something like this:
```C
   return log_full_errno(errno, quiet && errno == EPERM ? LOG_DEBUG : LOG_ERROR, ....);
```